src/data-access/repositories/mongo/mongo-action-repository.ts

```diff
@@ -0,0 +1,28 @@
+import { BaseMongoRepository } from './base-mongo-repository'
+import { ActionRepository } from '../interfaces/action-repository'
+import { Action } from '../../../model/entities/action'
+import { MongoDatabase } from './mongo-database'
+import { MongoEntityConverter } from './mongo-entity-converter'
+
+const COLLECTION_NAME: string = 'actions'
+
+export class MongoActionRepository extends BaseMongoRepository implements ActionRepository {
+
+  constructor(mongoDatabase: MongoDatabase, mongoEntityConverter: MongoEntityConverter) {
+    super(mongoDatabase, mongoEntityConverter)
+  }
+
+  protected getCollectionName(): string {
+    return COLLECTION_NAME
+  }
+
+  public async getAction(actionId: string): Promise<Action> {
+    return await this.getCollection().findOne({id: actionId}) as unknown as Action
+  }
+
+  public async saveActions(actions: Action[]): Promise<void> {
+    for (const action of actions) {
+      await this.getCollection().updateOne({ _id: action.id }, { $set: action }, { upsert: true })
+    }
```

Comment:
Yes, the `Array.map()` has some funky behavior when calling async methods

Reply:
Hmm, something very similar to what i suggested is being used in `mongo-test-database.ts`, and i havn't seen any issues during any of the tests that use it.
The file i mention does call `insertOne(...)` instead of the `.updateOne(...)`, but as both return a promise, and the issue you mention sound to be related to that, i don't see what that difference should make.

Can you give some more details as to what it does weirdly?

Reply:
@LindvedKrvang What funky behaviour are you referring to?
I don't see why the Promise.all approach, as @andr9528 suggested, shouldn't work. (Though the async and parentheses are unnecessary for the lambda :wink:)

Reply:
@KvelaGorrrrnio Something that you once told me, but I can't remember it. But if there is no issue, then I will update it

Reply:
Updated.